core/extension/gdextension_interface.h

```diff
@@ -2119,13 +2180,28 @@ typedef void (*GDExtensionInterfaceRefSetObject)(GDExtensionRefPtr p_ref, GDExte
  *
  * Creates a script instance that contains the given info and instance data.
  *
+ * This is deprecated. Use `script_instance_create2` instead.
```

Comment:
Should we have an attribute for that, that also conveys the information of when this was first deprecated?

Reply:
Great idea, would also make it machine-parseable.

Something that includes both version number when it was deprecated, and a message?
Just an idea:
```cpp 
/*
 * @deprecated 4.2 "Use `script_instance_create2` instead."
 */
```

Syntax can of course be bikeshedded, quotes may not be necessary. Would be allowed to span multiple lines, I guess.

Reply:
The doc comments in `gdextension_interface.cpp` are based on Doxygen which defines `@deprecated` (an alias for `\deprecated`) this way: https://www.doxygen.nl/manual/commands.html#cmddeprecated

If we follow that pattern, all the text after `@deprecated` is part of a paragraph that ends when the first blank comment line is encountered.

Perhaps we could use a sort of narrative style (where it's assumed the text starts with "Deprecated ...") like:

```c++
/**
 * @deprecated in Godot 4.2. Use `script_instance_create2` instead.
 */
```

I like this because it's a little more human readable and will hopefully be surfaced nicely by IDE tools (many of which understand Doxygen syntax).

The version number here is perhaps less machine-friendly than @Bromeon's suggestion, but if we adhere to this format strictly (and we'd need to be sure to adhere to any format strictly, including the previous suggestion) it can still be extracted by a parser.

Reply:
That sounds good. I have made the necessary documentation-comment changes.